src/include/gurt/common.h

```diff
@@ -58,8 +58,24 @@
 extern "C" {
 #endif
 
-#define likely(x)       __builtin_expect((x), 1)
-#define unlikely(x)     __builtin_expect((x), 0)
+#ifndef likely
+#define likely(x)		__builtin_expect((x), 1)
+#endif
+#ifndef unlikely
+#define unlikely(x)		__builtin_expect((x), 0)
+#endif
+
+#ifdef MPL_USE_MEMORY_TRACING
```

Comment:
this is really not the way we should be doing this.
we are adding a specific solution for MPICH here.
what if there are other libraries doing the same thing but have another macro for memory tracing?

I think all those memory macros and likely/unlikely should be just moved our of this public header into a private one.

Reply:
Unfortunately I don't see the way how to fix this is other way. MPICH require to use **MPL_calloc** instead of **calloc**. 
They have a following it they headers:
```
/* Define these as invalid C to catch their use in the code.
 * The ::: should cause the compiler to choke; the string will give the explanation
 */
#define malloc(a)         'Error use MPL_malloc' :::
#define calloc(a,b)       'Error use MPL_calloc' :::
#define free(a)           'Error use MPL_free'   :::
#define realloc(a)        'Error use MPL_realloc' :::
```
So, Which functions you propose to move into private header? Do you think we should use external d_alloc() instead of D_ALLOC macro?

Reply:
we should move all the D_*** macros out of this public header.
the same for likely/unlikely

Reply:
> Unfortunately I don't see the way how to fix this is other way. MPICH require to use **MPL_calloc** instead of **calloc**.

it requires using MPL_calloc for all code included in MPICH. which include this public header.
The problem is because we are including this in the mpich DAOS driver.
we should not have those macros that are only meant for internal use in a public header.